### PR TITLE
[FIX] website, *: remove number of subscribers on website form

### PR DESCRIPTION
**Pull request by SergeBayet**
*: website_mass_mailing

Before this commit, when a "Form" snippet was added and the action was
changed to "Subscribe to Newsletter", the mailing lists appeared as
checkbox fields with the number of subscribers in parentheses. This
commit removes the display of this unnecessary information.

Steps to reproduce:

- Install the "Email Marketing" module and Website.
- Navigate to the Website in edit mode.
- Drag & drop the "Form" block (dynamic content section).
- Change the form action by setting the "Action" option to "Subscribe to
  Newsletter".

Bug: The number of subscribers appears next to the mailing list names.

task-3472820

Co-authored-by: Adrien Milis <miad@odoo.com>

### CODE REVIEW

**Review by sobo-odoo — COMMENTED** [review]
Here is my pair-review :slightly_smiling_face:

File: addons/mass_mailing/models/mailing_list.py [hunk 1]
```diff
@@ -116,8 +116,9 @@ def write(self, vals):
 
         return super(MassMailingList, self).write(vals)
 
-    def name_get(self):
-        return [(list.id, "%s (%s)" % (list.name, list.contact_count)) for list in self]
+    def _compute_display_name(self):
+        for mailing_list in self:
+            mailing_list.display_name = f"{mailing_list.name} ({mailing_list.contact_count})"
```
[the remark this answers is not in the document]
Comment (SergeBayet) [reply addons/mass_mailing/models/mailing_list.py:122]: @sobo-odoo Good point. I'll make me a reminder to check the FW port on 16.4 and use the correct method.

**Comment by qsm-odoo** [comment]
Agreed with TDE @SergeBayet, this fix should only impact the snippet

**Comment by SergeBayet** [comment]
@tde-banana-odoo @qsm-odoo I've modified the code to avoid that. Thanks for your input.

**Review by reth-odoo — COMMENTED** [review]
I remember we concluded it wasn't cleanly doable 
https://github.com/odoo/odoo/pull/130876

But maybe you'll have better ideas

**Comment by SergeBayet** [comment]
@reth-odoo I guess my solution is not perfect regarding https://github.com/odoo/odoo/pull/130876. We'll see.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
@SergeBayet Is not that the same diff they made before? What was not working in theirs? In any case, let's try not to mention "mailing.list" in the "website" module :wink:

**Comment by qsm-odoo** [comment]
@SergeBayet I read the conversation in the task, I don't understand it. For me, if working, what was done in https://github.com/odoo/odoo/pull/130876 by @adrienmilis is exactly what we want (if both commits were squashed).

@awa-odoo @LNAodoo Can you confirm again why you canceled it? It seems it was because you did not like the fact that x2x fields are not dynamic in website forms but that's just the way it is at the moment (that may change in the future). Meanwhile it does seem nice to allow not to display the subscribers count for this specific field? Actually: the fact that it is not dynamic is an argument in favor of hiding that info, no?

**Comment by awa-odoo** [comment]
Hello @qsm-odoo 

From what I remember, it added complexity in the "form field" keys to support something we only need for a single use case.
It also does not fix existing labels, only the ones created post-fix (from what I understand).

It could be done still if you're ok with it, as it's a feedback that keeps coming back.

Hope that helps, cheers!

**Comment by LNAodoo** [comment]
Hello there, 

Indeed, from what I remember we canceled it because we considered the fix to be a bit too complicated.
But we're still for it so if there is an elegant solution, feel free to go for it :-)

Cheers!

**Comment by qsm-odoo** [comment]
@SergeBayet Let's take the work that was done in the other PR, move it here into one single commit, add you and the original dev as co-authors, make sure it works then we'll go with it. Thanks :pray:

**Comment by SergeBayet** [comment]
@qsm-odoo It's done. Hope that satisfies everyone.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
@SergeBayet 

- Drag & drop the "Form" block (dynamic content section).
- Change the form action by setting the "Action" option to "Subscribe to
  Newsletter".
- Remove the mailing list field (note that: should that be authorized? The form cannot be sent without that field so... should not it be model-required?)
- Re-add it (+ Field -> Existing field: Mailing lists)
=> Labels are blank :shrug:

**Comment by SergeBayet** [comment]
@qsm-odoo I fixed the bug concerning the blank labels and added the `modelRequired` constraint to the 'email' and 'list_ids' fields as in https://github.com/odoo/odoo/blob/16.0/addons/website/static/src/js/form_editor_registry.js#L32

**Review by qsm-odoo — APPROVED** [review]
Let's go with that, it's still not perfect as if I delete the field and re-add it, the numbers are back. But since we made the field required... the bug is technically hidden.

Would be easy enough to fix but I'd rather wait for the master owl refactor of this.

@robodoo r+

**Pull request closed without merging**